src/components/Categorize.js

```diff
@@ -0,0 +1,32 @@
+import React from 'react'
+import { Link,useLocation } from 'react-router-dom';
+
+
+const Categorize = ({categories,selectedCategory}) => {
+    const location = useLocation();
+  return (
+    <div className=" category">
+      {categories.map((category, i) =>
+        selectedCategory === category ? (
+          <Link
+            className="category__name category__name--selected"
+            key={i}
+            to={`/${location.pathname.split("/")[1]}/${category}`}
+          >
+            <div>{category}</div>
+          </Link>
+        ) : (
+          <Link
+            className="category__name"
+            key={i}
+            to={`/${location.pathname.split("/")[1]}/${category}`}
+          >
+            <div>{category}</div>
+          </Link>
+        )
+      )}
```

Comment:
```suggestion
      {categories.map((category, i) =>
        selectedCategory === category ? (
          <Link
            className="category__name category__name--selected"
            key={i}
            to={`/${location.pathname.split("/")[1]}/${category}`}
          >
            <div>{category}</div>
          </Link>
        ) : (
          <Link
            className="category__name"
            key={i}
            to={`/${location.pathname.split("/")[1]}/${category}`}
          >
            <div>{category}</div>
          </Link>
        )
      )}
```
this is fine for now , but in our codebase we use classNames package to assign multiple classname based on condition

Reply:
pathname as a variable could be extracted on top